app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -185,9 +186,12 @@ const MapImpl: React.FunctionComponent = () => {
         //loop through each feature type
         for (const featureType of Object.values(FeatureType)) {
           //Check if filters need to be added for this FeatureType
+          //We don't want to addFilters when some featuresByType lists are
+          //still dirty because then the filters will be removed by removeDataset() in the LOADED case
           if (
             state.featuresByType[featureType].featureTypeState ===
-            MapFeatureTypeState.NEEDS_FILTERS
+              MapFeatureTypeState.NEEDS_FILTERS &&
+            noDirtyFeaturesListCheck(state.featuresByType)
```

Comment:
Nit: that's a weird name for a function. hasDirtyFeatures ?

Reply:
You're only using this function in one place. Just declare it inline or as a top-level function (not exported) in Map.tsx?

In any case, this function should really be a one-liner, using Array.find() or the lodash (_).find. Something like:
!_.find(featuresByType, features => features.dirty).
https://lodash.com/docs/2.4.2#find